compiler/shared/test/diff/mono.mls

```diff
@@ -0,0 +1,1259 @@
+//│     fun isEmpty = () => {&&(((this).e1).isEmpty(), ((this).e2).isEmpty(),)}
+//│   }
+//│   Code(List((('(' new C$5([new Ch$3([1,]) {}, new A$4([new Ch$3([2,]) {}, new Ch$3([3,]) {},]) {},]) {} ')').derive(0,)).isEmpty()))
+//│ }
+//│ Mono:
+//│ 
+//│ Defunc result: 
+//│ main$$5()
+//│ fun isEmpty$C$5(this) =
+//│   &&(this.e1 match {case obj: Ep$2 => isEmpty$Ep$2(obj); case obj: E$1 => isEmpty$E$1(obj)}, this.e2 match {case obj: A$4 => isEmpty$A$4(obj)})
+//│ fun isEmpty$Ch$3(this) =
+//│   false
+//│ fun isEmpty$E$1(this) =
+//│   false
+//│ fun isEmpty$A$4(this) =
+//│   ||(this.e1 match {case obj: Ch$3 => isEmpty$Ch$3(obj)}, this.e2 match {case obj: Ch$3 => isEmpty$Ch$3(obj)})
+//│ fun derive$A$4(this, x) =
+//│   new A$4 (this.e1 match {case obj: Ch$3 => derive$Ch$3(obj, x)}, this.e2 match {case obj: Ch$3 => derive$Ch$3(obj, x)}) 
+//│ fun derive$C$5(this, x) =
+//│   if this.e1 match {case obj: Ch$3 => isEmpty$Ch$3(obj)} then new A$4 (new C$5 (this.e1 match {case obj: Ch$3 => derive$Ch$3(obj, x)}, this.e2) , this.e2 match {case obj: A$4 => derive$A$4(obj, x)})  else new C$5 (this.e1 match {case obj: Ch$3 => derive$Ch$3(obj, x)}, this.e2) 
+//│ fun derive$Ch$3(this, x) =
+//│   if ==(x, this.i) then new Ep$2 ()  else new E$1 () 
+//│ fun main$$5() =
+//│   new C$5 (new Ch$3 (#1) , new A$4 (new Ch$3 (#2) , new Ch$3 (#3) ) )  match {case obj: C$5 => derive$C$5(obj, #0)} match {case obj: C$5 => isEmpty$C$5(obj)}
+//│ fun isEmpty$Ep$2(this) =
+//│   true
+//│ class A$4(e1, e2) {
+//│ }
+//│ class E$1() {
+//│ }
+//│ class C$5(e1, e2) {
+//│ }
+//│ class Ep$2() {
+//│ }
+//│ class Ch$3(i) {
+//│ }
+//│ ╔══[ERROR] Class parameters currently need type annotations
+//│ ║  l.698: 	class Ch(i){
+//│ ╙──       	         ^
+//│ ╔══[ERROR] Class parameters currently need type annotations
+//│ ║  l.704: 	class A(e1, e2){
+//│ ╙──       	        ^^
+//│ ╔══[ERROR] Class parameters currently need type annotations
+//│ ║  l.704: 	class A(e1, e2){
+//│ ╙──       	            ^^
+//│ ╔══[ERROR] Class parameters currently need type annotations
+//│ ║  l.710: 	class C(e1, e2){
+//│ ╙──       	        ^^
+//│ ╔══[ERROR] Class parameters currently need type annotations
+//│ ║  l.710: 	class C(e1, e2){
+//│ ╙──       	            ^^
+//│ class E() {
+//│   fun derive: anything -> E
+//│   fun isEmpty: () -> false
+//│ }
+//│ class Ep() {
+//│   fun derive: anything -> E
+//│   fun isEmpty: () -> true
+//│ }
+//│ class Ch(i: error) {
+//│   fun derive: Num -> (E | Ep)
+//│   fun isEmpty: () -> false
+//│ }
+//│ class A(e1: error, e2: error) {
+//│   fun derive: anything -> A
+//│   fun isEmpty: () -> Bool
+//│ }
+//│ class C(e1: error, e2: error) {
+//│   fun derive: anything -> (A | C)
+//│   fun isEmpty: () -> Bool
+//│ }
+//│ Bool
+//│ res
+//│     = false
+
+
+:mono
+:e // FIXME: Malformed Test?
```

Comment:
@yuankeyu could you comment on this and propose a fixa? What were you trying to do?